### PR TITLE
Flav/oia tools ga

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

We've initiated the deprecation process for functions in favor of tools in our OpenAI implementation. This change has been available internally only, and we've observed no issues. This PR removes all the gates and delete the legacy function code. It extends the same logic to the Azure OpenAI version, as we were the only users. I've tested it, and tools are fully supported on Azure as well.

## Risk

Worst case, it breaks function call. Safe to rollback.

### CODE REVIEW

File: core/src/providers/openai.rs [hunk 1]
```diff
@@ -121,7 +121,7 @@ pub struct OpenAIToolFunction {
 }
```
Comment (flvndvd) [inline core/src/providers/openai.rs:121]: Diff is broken, I only removed `chat_completion` and `streamed_chat_completion` and rename the one with `_tools`.

**Review by spolu — APPROVED** [review]
Looks reasonable.

happy to test azure on Monday if you have not yet 👍

**Comment by flvndvd** [comment]
> Looks reasonable.
> 
> happy to test azure on Monday if you have not yet 👍

I did test Azure and it works like a charm!

**Comment by spolu** [comment]
\o/

**Pull request merged**